ttnn/cpp/ttnn/operations/moreh/moreh_group_norm/device/moreh_group_norm_device_operation.hpp

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_group_norm {
+struct MorehGroupNormOperation {
+    struct operation_attributes_t {
+        const uint32_t num_groups;
+        const float eps;
+        const std::vector<bool> are_required_outputs;
+        const MemoryConfig memory_config;
+        const MemoryConfig mean_memory_config;
+        const MemoryConfig rstd_memory_config;
+        const DeviceComputeKernelConfig compute_kernel_config;
+    };
+
+    struct tensor_args_t {
+        const Tensor& input;
+        const std::optional<const Tensor> gamma;
+        const std::optional<const Tensor> beta;
+        const std::vector<std::optional<Tensor>> output_tensors;
```

Comment:
wouldn't it be better if instead of vector (with unknown length) this were 2 (or 3?) optional args?

Reply:
You're right. I did that originally but decided to change it for better clarity. I've reverted it back to 2 (or 3) optional args.